macosx/QuickLookExtension/PreviewViewController.mm

```diff
@@ -0,0 +1,146 @@
+// This file Copyright © 2010-2022 Transmission authors and contributors.
+// It may be used under the MIT (SPDX: MIT) license.
+// License text can be found in the licenses/ folder.
+
+#import "PreviewViewController.h"
+
+#include <libtransmission/error.h>
+#include <libtransmission/torrent-metainfo.h>
+
+#import "NSStringAdditions.h"
+
+#import <QuickLookUI/QLPreviewingController.h>
+
+#pragma mark - TRError
+
+@interface TRError : NSError
+
+- (instancetype)initWith:(tr_error*)error;
+
+@end
+
+@interface TRError ()
+
+@property NSString* errorMessage;
+
+@end
+
+@implementation TRError
+
+- (instancetype)initWith:(tr_error*)error
+{
+    if (self = [super initWithDomain:@"TransmissionError" code:error->code userInfo:nil])
+    {
+        self.errorMessage = @(error->message);
+    }
+    return self;
+}
+
+- (NSString*)localizedDescription
+{
+    return self.errorMessage;
+}
+
+@end
+
+#pragma mark -
+
+@interface PreviewViewController ()<QLPreviewingController>
+
+#pragma mark - IBOutlet
+@property(weak) IBOutlet NSImageView* fileIconImage;
+@property(weak) IBOutlet NSTextField* nameField;
+@property(weak) IBOutlet NSTextField* totalSizeField;
+@property(weak) IBOutlet NSTextField* filesField;
+@property(weak) IBOutlet NSTextField* trackersField;
+
+@end
+
+#pragma mark -
+
+@implementation PreviewViewController
+
+#pragma mark - NSViewController
+
+- (NSString*)nibName
+{
+    return @"PreviewViewController";
+}
+
+- (void)loadView
+{
+    [super loadView];
```

Comment:
Is this required? The docs say starting from 10.10 this method and the one above can be omitted as long as the nib name is the same (which it is)

Reply:
I'll test without it and remove if really it is not needed anymore.